### ISSUE TITLE
Versioning and React dependency strategy?

### ISSUE THREAD

**Issue by Ephem** [Ephem/react#1]
Currently Lightyear is versioned as semver with the following dependencies:

```json
{
  "version": "0.1.0",
  "peerDependencies": {
    "react": "^16.0.0"
  },
  "optionalDependencies": {
    "react-dom": "16.8.6"
  }
}
```

It would be nice if Lightyear versions aligned with the React-versions (just as `react` and `react-dom` are aligned), however, I see two potential issues I'm not sure about:

* How to release bug fixes for Lightyear while retaining alignment with React versions? (Or for that matter majors or minors if functionality like fallbacks are added?)
* Lightyear touches a part of the React-codebase that is not updated very often (right now). Many releases might just be bumps to align with React. Not much of an issue, just an observation. :)

One idea could be to sync major and minors, but retain patch-versions for bumps to Lightyear itself?

Would love feedback on this!

**Comment by voxpelli** [comment]
I would suggest the same as you: Keep sync on major and minor, have patch for yourself.

Also:

Are you sure about the `optionalDependencies`? That will not be a peer dependency but will, if mismatched to the top level `react-dom`, be a duplicate of that specific version within this library itself. Is that the intention?

And why have it as an `optionalDependencies`?

That pretty much only have an effect on the dependencies when the environment isn't compatible with it – like a native package that's only compatible with Linux, not with eg. Windows – then it won't block an install, it will just fail silently.

Can that ever happen with `react-dom`?

(Interesting project btw 👍 Do you use this internally at Pricerunner? We're also a heavy React SSR user at @Sydsvenskan / HD.se, this might be something for us to look into as well and possibly even maybe contribute something to – if we find some need or bug or such 😊)

**Comment by Ephem** [comment]
Thank you for the feedback! I am a little worried it might get confusing, but I think that might be a good alternative. 😄

Good catch with the `optionalDependencies`, that is indeed not at all what I intended. I haven't used it before and think I read the docs entry a bit quickly.. 😞 `react-dom` isn't really a dependency of Lightyear at all since client hydration is optional, but when it is used it should match versions. I guess this is still is a kind of `peerDependency`..?

Thank you! So far this is purely a personal hobby project, but we are looking at refactoring our data fetching as well as our server rendering setup at Pricerunner and I have been trying this out there, so who knows? 😄

Would definitely love contributions! Right now just having people experiment with it would be perfect. 👍

**Comment by Ephem** [comment]
I just released `v0.1.1` which moves `react-dom` to be a `peerDependency` instead of an optional one, thanks again for reporting!

When it comes to versioning I'm still uncertain, but I am still leaning towards syncing major/minor. I will hold off on this for now though, at least until the project is more stable since any breaking changes to Lightyear itself after such a versioning change would have to break semver and only bump patch.
